static/app/components/nav/utils.tsx

```diff
@@ -0,0 +1,165 @@
+}
+
+export enum ActiveStatus {
+  INACTIVE = 'inactive',
+  ACTIVE_PARENT = 'active-parent',
+  ACTIVE = 'active',
+}
+
+export function getActiveStatus(
+  item: SidebarItem | SubmenuItem,
+  location: ReturnType<typeof useLocation>
+): ActiveStatus {
+  if (item.to.includes('/issues/') && item.to.includes('query=')) {
+    if (location.search.includes('viewId') && item.label === 'All')
+      return ActiveStatus.ACTIVE;
+    return hasMatchingQueryParam({to: item.to, label: item.label}, location)
+      ? ActiveStatus.ACTIVE
+      : ActiveStatus.INACTIVE;
+  }
+  const normalizedTo = normalizeUrl(item.to);
+  const normalizedCurrent = normalizeUrl(location.pathname);
+  if (normalizedTo === normalizedCurrent) return ActiveStatus.ACTIVE;
+  if (isItemActive({to: item.to, label: item.label})) return ActiveStatus.ACTIVE;
+  if (
+    'submenu' in item &&
+    item.submenu?.find(
+      subitem =>
+        typeof subitem === 'object' &&
+        getActiveStatus(subitem, location) !== ActiveStatus.INACTIVE
+    ) !== undefined
+  ) {
+    return ActiveStatus.ACTIVE_PARENT;
+  }
+  return ActiveStatus.INACTIVE;
+}
+
+export function getActiveProps(
+  status: ActiveStatus
+): Partial<Pick<HTMLProps<HTMLElement>, 'className' | 'aria-current'>> {
+  switch (status) {
+    case ActiveStatus.ACTIVE_PARENT:
+      return {className: 'active'};
+    case ActiveStatus.ACTIVE:
+      return {className: 'active', 'aria-current': 'page'};
+    case ActiveStatus.INACTIVE:
+    default:
+      return {};
+  }
+}
+
+export function hasMatchingQueryParam(
+  item: Pick<SidebarItem | SubmenuItem, 'to' | 'label'>,
+  location: ReturnType<typeof useLocation>
+): boolean {
+  if (location.search.length === 0) return false;
+  if (item.to.includes('?')) {
+    const search = new URLSearchParams(location.search);
+    const itemSearch = new URLSearchParams(item.to.split('?').at(-1));
+    const itemQuery = itemSearch.get('query');
+    const query = search.get('query');
+    if (item?.label === 'All') {
+      return !query && !itemQuery;
+    }
+    if (itemQuery && query) {
+      let match = false;
+      for (const key of itemQuery?.split(' ')) {
+        match = query.includes(key);
+        if (!match) continue;
+      }
+      return match;
+    }
+  }
+  return false;
+}
+
+export function splitAtDivider<T>(arr: (T | typeof NAV_DIVIDER)[]): {
+  body: T[];
+  footer: T[];
+} {
+  const body: T[] = [];
+  const footer: T[] = [];
+  let current = body;
+  for (const item of arr) {
+    if (item === NAV_DIVIDER) {
+      current = footer;
+      continue;
+    }
+    current.push(item);
+  }
+  return {body, footer};
+}
+
+export function useLocationDescriptor(to?: string): LocationDescriptor {
+  return useMemo(() => {
+    if (!to) {
+      return '#';
+    }
+    const [pathname, search] = to.split('?');
+
+    return {
+      pathname,
+      search: search ? `?${search}` : undefined,
+      state: {source: SIDEBAR_NAVIGATION_SOURCE},
+    };
+  }, [to]);
+}
```

Comment:
I dont think this needs to be a hook, the naming looks like something we would expose from react router and it's really doing a conversion from a to obj.  How about something like makeLocationDescriptorFromTo or something similar and just expose that, wrapping it inside a useMemo is trivial for consumers to add

Reply:
Good catch—no need for this to be a hook! Refactored and renamed in [3bb7468](https://github.com/getsentry/sentry/pull/77665/commits/3bb74680d614a2e6f6f17cbdfbf2af5dc4c08a83).